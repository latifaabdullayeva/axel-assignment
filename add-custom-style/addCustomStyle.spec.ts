import StyleDecorator from "./addCustomStyle";

describe("style decorator", () => {
  const styleDecorator = new StyleDecorator();

  afterEach(() => {
    // Unfortunately, jest does not reset the state of the DOM between
    // test runs. Hence, we have to do it ourselves :(
    document.body.innerHTML = "";
  });

  test("creates a style without mediaquery", () => {
    styleDecorator.addCustomStyle(
      "myStyle",
      ".someClassName",
      "{background-color: #cccccc;}"
    );
    expect(document.body).toMatchSnapshot();
  });

  test("creates a style with a provided mediaQuery", () => {
    styleDecorator.addCustomStyle(
      "myStyle",
      ".someClassName",
      "{background-color: #cccccc;}",
      "max-width: 909px"
    );

    expect(document.body).toMatchSnapshot();
  });

  test("fails if any mandatory argument is missing", () => {    
    expect(() => {       
      styleDecorator.addCustomStyle(undefined, "two", "three");
    }).toThrowError();
    expect(() => {
      styleDecorator.addCustomStyle("one", undefined, "three");
    }).toThrowError();
    expect(() => {
      styleDecorator.addCustomStyle("one", "two", undefined);
    }).toThrowError();
  });

  test("fails if any of mandatory arguments are present but empty", () => {
    expect(() => {
      styleDecorator.addCustomStyle("", "two", "three");
    }).toThrowError();
    expect(() => {
      styleDecorator.addCustomStyle("one", "", "three");
    }).toThrowError();
    expect(() => {
      styleDecorator.addCustomStyle("one", "two", "");
    }).toThrowError();
  });

  test("fails if brackets around attributes are missing", () => {
    expect(() => {
      styleDecorator.addCustomStyle(
        "myStyle",
        ".someClassName",
        "background-color: #cccccc;"
      );
    }).toThrowError();
  });

  test("attributes surrounded by space must still be valid", () => {
    styleDecorator.addCustomStyle(
      "myStyle",
      ".someClassName",
      "  {background-color: #cccccc;} "
    );
    expect(document.body).toMatchSnapshot();
  });
});
