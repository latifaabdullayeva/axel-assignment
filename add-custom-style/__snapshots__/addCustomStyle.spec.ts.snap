// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style decorator attributes surrounded by space must still be valid 1`] = `
<body>
  <style
    id="myStyle"
  >
    .someClassName  {background-color: #cccccc;} 
  </style>
</body>
`;

exports[`style decorator creates a style with a provided mediaQuery 1`] = `
<body>
  <style
    id="myStyle"
  >
    @media (max-width: 909px){.someClassName{background-color: #cccccc;}}
  </style>
</body>
`;

exports[`style decorator creates a style without mediaquery 1`] = `
<body>
  <style
    id="myStyle"
  >
    .someClassName{background-color: #cccccc;}
  </style>
</body>
`;
